pravega/src/main/scala/akka/stream/alpakka/pravega/settings.scala

```diff
@@ -0,0 +1,246 @@
+  def readerId = settings.readerId
+  def readerConfig = settings.readerConfig
+  def clientConfig = settings.clientConfig
+  def timeout = settings.timeout
+}
+
+object ReaderSettingsBuilder {
+  val configPath = "akka.alpakka.pravega"
+
+  /**
+   * Create reader settings from the default configuration
+   * `akka.alpakka.pravega`.
+   */
+  def apply[A](actorSystem: ActorSystem): ReaderSettingsBuilder =
+    apply(actorSystem.settings.config.getConfig(configPath))
+
+  /**
+   * Create reader settings from the default configuration
+   * `akka.alpakka.pravega`.
+   * @param clientConfigBuilder customisation function
+   */
+  def apply[A](actorSystem: ActorSystem, clientConfigBuilder: ClientConfigBuilder => Unit): ReaderSettingsBuilder =
+    apply(actorSystem.settings.config.getConfig(configPath), Some(clientConfigBuilder))
+
+  /**
+   * Java API: Create reader settings from the default configuration
+   * `akka.alpakka.pravega`.
+   */
+  def create[A](actorSystem: ActorSystem): ReaderSettingsBuilder =
+    apply(actorSystem.settings.config.getConfig(configPath))
+
+  /**
+   * Java API: Create reader settings from the default configuration
+   * `akka.alpakka.pravega`.
+   * @param clientConfigBuilder customisation function
+   */
+  def create[A](actorSystem: ActorSystem, clientConfigBuilder: Consumer[ClientConfigBuilder]): ReaderSettingsBuilder =
+    apply(actorSystem.settings.config.getConfig(configPath), Some(clientConfigBuilder.asScala))
+
+  /**
+   * Create settings from a configuration with the same layout as
+   * the default configuration `akka.alpakka.pravega.reader`.
+   */
+  def apply[A](config: Config,
+               clientConfigBuilder: Option[ClientConfigBuilder => Unit] = None): ReaderSettingsBuilder = {
+
+    import ConfigHelper._
+
+    implicit val rConfig = config.getConfig("reader")
+
+    val readerBasicSetting = new ReaderBasicSetting()
+
+    extractString("group-name")(readerBasicSetting.withGroupName)
+    extractString("reader-id")(readerBasicSetting.withReaderId)
+    extractLong("timeout")(readerBasicSetting.withTimeout)
+
+    val clientConfig = ConfigHelper.buildClientConfig(config, "client-config", clientConfigBuilder)
+
+    val readerConfig = ConfigHelper.buildReaderConfig(rConfig, "config")
+
+    new ReaderSettingsBuilder(clientConfig,
+                              readerConfig,
+                              readerBasicSetting.groupName,
+                              readerBasicSetting.readerId,
+                              readerBasicSetting.timeout)
+
+  }
+}
+
+class WriterSettingsBuilder(val clientConfig: ClientConfig,
+                            val eventWriterConfig: EventWriterConfig,
+                            maximumInflightMessages: Int) {
+  def withSerializer[A](serializer: Serializer[A]): WriterSettings[A] =
+    new WriterSettings[A](this, serializer, None, maximumInflightMessages)
+  def withSerializer[A](serializer: Serializer[A], extractor: A => String) =
```

Comment:
```suggestion
  def withSerializer[Message](serializer: Serializer[Message], keyExtractor: Message => String) =
```

I replaced 2 things here: `extractor` to `keyExtractor` and `A` to `Message`. I think the second one could be extended to other parts of the code, I generally like meaningful names for types and try to avoid `T` or `R`, etc... but that's a matter of taste.

Reply:
Haha, with had this discussion at office today ... But as you say, matter of taste, even https://docs.scala-lang.org/style/naming-conventions.html#type-parameters-generics not interpreted the same way by everybody ;)